## Author Chinyere Obi

In [1]:
import pandas as pd  # import the library

pd.set_option('display.max_columns',None)  # this displays the maximum columns in the dataframe

In [2]:
df = pd.read_csv(r"C:\Users\hp\Desktop\TSF-Internship\globalterrorism.csv", encoding="ISO-8859-1")
df.head()                      

C:\Users\hp\AppData\Local\Temp\ipykernel_9620\4073430860.py:1: DtypeWarning: Columns (4,6,31,33,61,62,63,76,79,90,92,94,96,114,115,121) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(r"C:\Users\hp\Desktop\TSF-Internship\globalterrorism.csv", encoding="ISO-8859-1")


eventid  iyear  imonth  iday approxdate  extended resolution  country  \
0  197000000001   1970       7     2        NaN         0        NaN       58   
1  197000000002   1970       0     0        NaN         0        NaN      130   
2  197001000001   1970       1     0        NaN         0        NaN      160   
3  197001000002   1970       1     0        NaN         0        NaN       78   
4  197001000003   1970       1     0        NaN         0        NaN      101   

          country_txt  region                   region_txt provstate  \
0  Dominican Republic       2  Central America & Caribbean       NaN   
1              Mexico       1                North America   Federal   
2         Philippines       5               Southeast Asia    Tarlac   
3              Greece       8               Western Europe    Attica   
4               Japan       4                    East Asia   Fukouka   

            city   latitude   longitude  specificity  vicinity location  \
0  Santo Domingo  18.456792  -69.951164          1.0         0      NaN   
1    Mexico city  19.371887  -99.086624          1.0         0      NaN   
2        Unknown  15.478598  120.599741          4.0         0      NaN   
3         Athens  37.997490   23.762728          1.0         0      NaN   
4        Fukouka  33.580412  130.396361          1.0         0      NaN   

  summary  crit1  crit2  crit3  doubtterr  alternative alternative_txt  \
0     NaN      1      1      1        0.0          NaN             NaN   
1     NaN      1      1      1        0.0          NaN             NaN   
2     NaN      1      1      1        0.0          NaN             NaN   
3     NaN      1      1      1        0.0          NaN             NaN   
4     NaN      1      1      1       -9.0          NaN             NaN   

   multiple  success  suicide  attacktype1                 attacktype1_txt  \
0       0.0        1        0            1                   Assassination   
1       0.0        1        0            6     Hostage Taking (Kidnapping)   
2       0.0        1        0            1                   Assassination   
3       0.0        1        0            3               Bombing/Explosion   
4       0.0        1        0            7  Facility/Infrastructure Attack   

   attacktype2 attacktype2_txt  attacktype3 attacktype3_txt  targtype1  \
0          NaN             NaN          NaN             NaN         14   
1          NaN             NaN          NaN             NaN          7   
2          NaN             NaN          NaN             NaN         10   
3          NaN             NaN          NaN             NaN          7   
4          NaN             NaN          NaN             NaN          7   

                 targtype1_txt  targsubtype1  \
0  Private Citizens & Property          68.0   
1      Government (Diplomatic)          45.0   
2          Journalists & Media          54.0   
3      Government (Diplomatic)          46.0   
4      Government (Diplomatic)          46.0   

                                    targsubtype1_txt  \
0                                     Named Civilian   
1  Diplomatic Personnel (outside of embassy, cons...   
2                    Radio Journalist/Staff/Facility   
3                                  Embassy/Consulate   
4                                  Embassy/Consulate   

                         corp1                  target1  natlty1  \
0                          NaN             Julio Guzman     58.0   
1  Belgian Ambassador Daughter  Nadine Chaval, daughter     21.0   
2             Voice of America                 Employee    217.0   
3                          NaN             U.S. Embassy    217.0   
4                          NaN           U.S. Consulate    217.0   

          natlty1_txt  targtype2 targtype2_txt  targsubtype2 targsubtype2_txt  \
0  Dominican Republic        NaN           NaN           NaN              NaN   
1             Belgium        NaN           NaN           NaN              NaN   


In [3]:
# Shows the total rows and columns of the data

df.shape

(181691, 135)

In [4]:
# This list all column names of the data

df.columns.to_list()

['eventid',
 'iyear',
 'imonth',
 'iday',
 'approxdate',
 'extended',
 'resolution',
 'country',
 'country_txt',
 'region',
 'region_txt',
 'provstate',
 'city',
 'latitude',
 'longitude',
 'specificity',
 'vicinity',
 'location',
 'summary',
 'crit1',
 'crit2',
 'crit3',
 'doubtterr',
 'alternative',
 'alternative_txt',
 'multiple',
 'success',
 'suicide',
 'attacktype1',
 'attacktype1_txt',
 'attacktype2',
 'attacktype2_txt',
 'attacktype3',
 'attacktype3_txt',
 'targtype1',
 'targtype1_txt',
 'targsubtype1',
 'targsubtype1_txt',
 'corp1',
 'target1',
 'natlty1',
 'natlty1_txt',
 'targtype2',
 'targtype2_txt',
 'targsubtype2',
 'targsubtype2_txt',
 'corp2',
 'target2',
 'natlty2',
 'natlty2_txt',
 'targtype3',
 'targtype3_txt',
 'targsubtype3',
 'targsubtype3_txt',
 'corp3',
 'target3',
 'natlty3',
 'natlty3_txt',
 'gname',
 'gsubname',
 'gname2',
 'gsubname2',
 'gname3',
 'gsubname3',
 'motive',
 'guncertain1',
 'guncertain2',
 'guncertain3',
 'individual',
 'nperps',
 'nperpcap',
 

In [5]:
# columns that will be used in the analysis

df = df[["iyear","country_txt","region_txt","provstate",
         "suicide","attacktype1_txt","targtype1_txt",
          "gname","weaptype1_txt","nkill","nwound"]]
df.head()

iyear         country_txt                   region_txt provstate  suicide  \
0   1970  Dominican Republic  Central America & Caribbean       NaN        0   
1   1970              Mexico                North America   Federal        0   
2   1970         Philippines               Southeast Asia    Tarlac        0   
3   1970              Greece               Western Europe    Attica        0   
4   1970               Japan                    East Asia   Fukouka        0   

                  attacktype1_txt                targtype1_txt  \
0                   Assassination  Private Citizens & Property   
1     Hostage Taking (Kidnapping)      Government (Diplomatic)   
2                   Assassination          Journalists & Media   
3               Bombing/Explosion      Government (Diplomatic)   
4  Facility/Infrastructure Attack      Government (Diplomatic)   

                                gname weaptype1_txt  nkill  nwound  
0                              MANO-D       Unknown    1.0     0.0  
1  23rd of September Communist League       Unknown    0.0     0.0  
2                             Unknown       Unknown    1.0     0.0  
3                             Unknown    Explosives    NaN     NaN  
4                             Unknown    Incendiary    NaN     NaN

In [6]:
#Renaming the columns

df.rename(columns={"iyear":"Year","country_txt":"Country",
                   "region_txt":"Region","provstate":"State",
                   "attacktype1_txt":"Type of Attack","targtype1_txt":"Target Type",
                   "gname":"Name of Group","weaptype1_txt":"Type of Weapon",
                   "nkill":"Killed","nwound":"Wounded", 'suicide':'Suicide'},inplace=True)

In [7]:
# Provides a concise summary of the dataframe

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 181691 entries, 0 to 181690
Data columns (total 11 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   Year            181691 non-null  int64  
 1   Country         181691 non-null  object 
 2   Region          181691 non-null  object 
 3   State           181270 non-null  object 
 4   Suicide         181691 non-null  int64  
 5   Type of Attack  181691 non-null  object 
 6   Target Type     181691 non-null  object 
 7   Name of Group   181691 non-null  object 
 8   Type of Weapon  181691 non-null  object 
 9   Killed          171378 non-null  float64
 10  Wounded         165380 non-null  float64
dtypes: float64(2), int64(2), object(7)
memory usage: 15.2+ MB


In [8]:
# Replace null values with 0

df['Killed'] = df['Killed'].fillna(0)
df['Wounded'] = df['Wounded'].fillna(0)

In [9]:
# Add a column for casualty

df['Casualty'] = df['Killed'] + df['Wounded']
df.head()

Year             Country                       Region    State  Suicide  \
0  1970  Dominican Republic  Central America & Caribbean      NaN        0   
1  1970              Mexico                North America  Federal        0   
2  1970         Philippines               Southeast Asia   Tarlac        0   
3  1970              Greece               Western Europe   Attica        0   
4  1970               Japan                    East Asia  Fukouka        0   

                   Type of Attack                  Target Type  \
0                   Assassination  Private Citizens & Property   
1     Hostage Taking (Kidnapping)      Government (Diplomatic)   
2                   Assassination          Journalists & Media   
3               Bombing/Explosion      Government (Diplomatic)   
4  Facility/Infrastructure Attack      Government (Diplomatic)   

                        Name of Group Type of Weapon  Killed  Wounded  \
0                              MANO-D        Unknown     1.0      0.0   
1  23rd of September Communist League        Unknown     0.0      0.0   
2                             Unknown        Unknown     1.0      0.0   
3                             Unknown     Explosives     0.0      0.0   
4                             Unknown     Incendiary     0.0      0.0   

   Casualty  
0       1.0  
1       0.0  
2       1.0  
3       0.0  
4       0.0

In [10]:
# Check for missing values

df.isnull().sum()

Year                0
Country             0
Region              0
State             421
Suicide             0
Type of Attack      0
Target Type         0
Name of Group       0
Type of Weapon      0
Killed              0
Wounded             0
Casualty            0
dtype: int64

In [11]:
# Provides a statistical summary of the data

df.describe()

Year        Suicide         Killed        Wounded  \
count  181691.000000  181691.000000  181691.000000  181691.000000   
mean     2002.638997       0.036507       2.266860       2.883296   
std        13.259430       0.187549      11.227057      34.309747   
min      1970.000000       0.000000       0.000000       0.000000   
25%      1991.000000       0.000000       0.000000       0.000000   
50%      2009.000000       0.000000       0.000000       0.000000   
75%      2014.000000       0.000000       2.000000       2.000000   
max      2017.000000       1.000000    1570.000000    8191.000000   

            Casualty  
count  181691.000000  
mean        5.150156  
std        40.555416  
min         0.000000  
25%         0.000000  
50%         1.000000  
75%         4.000000  
max      9574.000000

## Observations:
The period of terrorist activities were from 1970 - 2017
The Maximum number of people killed were 1570
The Maximum number of people wounded were 8191
The Maximum number of casualties were 9574.

In [12]:
df.to_csv(r'C:\Users\hp\Desktop\TSF-Internship\Global_Terrorism.csv', index=False) 